lib/index.js

```diff
@@ -263,16 +312,72 @@ api.get = async ({actor, account, email, type} = {}) => {
   } else {
     query['account.email'] = email;
   }
+
   const record = await database.collections.account.findOne(
     query, {projection: {_id: 0, [typeKey]: 1}});
+
   if(!record) {
     throw new BedrockError('Authentication token not found.', 'NotFoundError', {
       httpStatusCode: 404,
       public: true
     });
   }
 
-  return record.meta['bedrock-authn-token'].tokens[type];
+  const result = record.meta['bedrock-authn-token'].tokens[type];
+  if(type === 'nonce') {
+    if(tokenId) {
+      return result.filter(token => token.id === tokenId)[0];
+    } else {
```

Comment:
Unnecessary `else` here and below (because the `if` above does a `return` already).